# 1. Loading the dataset using pandas

In [1]:
import pandas as pd

# Setting the columns for the dataset
cols = ['age', 'workclass', 'fnlwgt', 
        'education', 'education-num', 
        'marital-status', 'occupation', 
        'relationship', 'race', 'sex', 
        'capital-gain', 'capital-loss', 
        'hours-per-week', 'native-country', 
        'income']

# Reading the dataset and storing into adult
adults = pd.read_csv('./Adults/adult.data', names = cols, index_col = False)

# 2. Exploring and analysing the dataset

### Firstly, let us have a rough idea about the dataset

In [2]:
adults.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   fnlwgt          32561 non-null  int64 
 3   education       32561 non-null  object
 4   education-num   32561 non-null  int64 
 5   marital-status  32561 non-null  object
 6   occupation      32561 non-null  object
 7   relationship    32561 non-null  object
 8   race            32561 non-null  object
 9   sex             32561 non-null  object
 10  capital-gain    32561 non-null  int64 
 11  capital-loss    32561 non-null  int64 
 12  hours-per-week  32561 non-null  int64 
 13  native-country  32561 non-null  object
 14  income          32561 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


From the .info() method, we can see there are a total of **14 attributes** and **1 output label** which is **income**.

Let's discuss some observations below:

1. Out of the **14 attributes**, 6 of it are numerical data, whereas the remaining 8 are categorical data.
2. There are a total of **32561 instances**

### Now let us view the distribution of the output label

In [3]:
# Lets call the income as y for simplicity sake

adults.income.value_counts()

 <=50K    24720
 >50K      7841
Name: income, dtype: int64